### DISCUSSION TITLE
Suggestion on improving PSO optimization.

### DISCUSSION THREAD

**Discussion by mamun-nahid** [AstroBridge/BDLensing#158] [Q&A] [answered]
Can anyone suggest how I can model the marked regions properly? Also, how can I remove the residuals at the centre? 
You can find the notebook [here.](https://colab.research.google.com/drive/1eXg7Zq7kmGkKUC06_dp5R6MC-ctX3RlR?usp=sharing)

![Untitled](https://github.com/ajshajib/BDLensing/assets/61522619/580ab9da-1aba-49e2-80b3-8848cfe82bf1)

**Comment by rafee-adnan** [comment, marked as answer]
I think a smaller `beta` value will fix it. My intuition tells me that this will be fixed if you start your MCMC run. It will converge to a lower value. (I might be wrong 😅)

**Reply by mamun-nahid** [reply]
Okay. I will check whether lowering the max value of `beta` improves it or not.  If it does not work I try with MCMC. I will let you know. Thank you very much.

**Reply by mamun-nahid** [reply]
I have tried setting beta like this:

```
kwargs_source_init.append({"beta": 0.0, "center_x": 0.0, "center_y": 0.0})
kwargs_source_sigma.append({"beta": 0.001, "center_x": 0.2, "center_y": 0.2})
kwargs_lower_source.append({"beta": 0.001, "center_x": -10, "center_y": -10})
kwargs_upper_source.append({"beta": 0.08, "center_x": 10, "center_y": 10})
```
It says Converged after 74 iterations and it does not reconstruct anything. 

Should I start MCMC?

**Reply by rafee-adnan** [reply]
Your initial values must lie in between the lower limit and upper limit. You can not initialize `beta` at 0.

**Reply by mamun-nahid** [reply]
Yes. I have used the initial value of 0.02. I think the lower arc is a little better now but the upper arc is almost the same. I am not sure which one is better. The reconstructed source looks different.
notebook [link](https://colab.research.google.com/drive/1Ss3ZmiVULSlyIWF5o84laamenwiU0X1q?usp=sharing)

![r5](https://github.com/ajshajib/BDLensing/assets/61522619/40d2d0a1-a376-4619-87fc-0d41de443f0f)
![r5_2](https://github.com/ajshajib/BDLensing/assets/61522619/ea3f169b-cfe6-4a7b-9086-f50fd4abe353)

**Reply by rafee-adnan** [reply]
I think your fit for the marked regions of both the upper and lower arc got better than before. The blue region decreased in both of them.

**Reply by mamun-nahid** [reply]
Yeah. Does the reconstructed source look alright? I can decide which one is better.

**Reply by rafee-adnan** [reply]
Yes, it looks alright to me. You don't have to worry much regarding the reconstructed source as you have already got a decent fit. Best wishes!

**Reply by mamun-nahid** [reply]
Okay bhai. Thank you very much.
